api/v2/checluster_types.go

```diff
@@ -337,9 +337,6 @@ type ImagePuller struct {
 // See https://github.com/devfile/devworkspace-operator
 // +k8s:openapi-gen=true
 type DevWorkspace struct {
-	// Deployment override options.
-	// +optional
-	Deployment *Deployment `json:"deployment,omitempty"`
```

Comment:
To double check -- this field is intended only for use on Kubernetes, right? 

We had a discussion earlier about removing CRD fields and whether OLM would permit the change. We figured it was generally fine at the time but in further testing I've found that OLM blocks operator upgrades if the new CRD would render any existing CR invalid, meaning that if anyone has an OLM install and has set this field, we would need to document that this field has to be dropped before the upgrade can take place.

Reply:
Yes, it was just for Kubernetes.
I see, do I unserstand correctly that you suggest keeping this field for a while?

Reply:
I'm not sure the best way forward -- probably the best approach is to just remove it, assuming not many users have configured the field in their CheCluster CRs. If the field isn't set in the CheCluster, then the update can proceed with no issues. The bigger issue is OLM where the installPlan would fail, requiring deleting + reinstalling the Operator, but that's extremely unlikely since it's a field meant for Kubernetes.

If we do get reports on failing to upgrade in the future, we should keep this in mind as a potential cause. The fix is relatively simple (delete the field before attempting the install).